### ISSUE TITLE
[BUG]: Fails to start

### ISSUE THREAD

**Issue by pablorq** [sagetarian/skype-wrapper#26]
I've follwed the install guide from OMG Ubuntu but it didn't work. (http://www.omgubuntu.co.uk/2011/10/how-to-add-skype-to-the-ubuntu-messaging-menu/)

Installed: 
- python-skype: 1.0.31.0-2
- skype-wrapper_0ubuntu5-3_all.deb

Skype-wrapper asks for permission, but then never starts in the messaging menu.

Added the PPA and updated to skype-wrapper 0ubuntu5-4. Same behavior.

Deleted permission inside skype and runned again. Same behavior. 

My system: 
Ubuntu 11.10
Skype 2.2.0.35
PPA for skype-wrapper
Telepathy uninstalled
Pidgin + pidgin-skype plugin

Terminal output:
$ skype-wrapper
Starting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion`g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
Traceback (most recent call last):
  File "indicator-applet-skype.py", line 832, in <module>
    skype = SkypeBehaviour();
  File "indicator-applet-skype.py", line 481, in __init__
    self.telepathy_presence = self.getPresence()
  File "indicator-applet-skype.py", line 793, in getPresence
    '/org/freedesktop/Telepathy/AccountManager')
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 244, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python2.7/dist-packages/dbus/proxies.py", line 246, in **init**
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 183, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 281, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python2.7/dist-packages/dbus/connection.py", line 630, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Telepathy.AccountManager was not provided by any .service files
Applet closed
API crash detected
Restarting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
Traceback (most recent call last):
  File "indicator-applet-skype.py", line 832, in <module>
    skype = SkypeBehaviour();
  File "indicator-applet-skype.py", line 460, in __init__
    self.skype.Client.Minimize()
  File "/usr/lib/pymodules/python2.7/Skype4Py/client.py", line 98, in Minimize
    self._Skype._DoCommand('MINIMIZE')
  File "/usr/lib/pymodules/python2.7/Skype4Py/skype.py", line 359, in _DoCommand
    self.SendCommand(command)
  File "/usr/lib/pymodules/python2.7/Skype4Py/skype.py", line 801, in SendCommand
    self._API.SendCommand(Command)
  File "/usr/lib/pymodules/python2.7/Skype4Py/API/posix_x11.py", line 405, in SendCommand
    raise ISkypeAPIError('Skype command timeout')
Skype4Py.errors.ISkypeAPIError: Skype command timeout
Applet closed

Workaround form other issue:
$ kill indicator-skype
bash: kill: indicator-skype: los argumentos deben ser procesos o IDs de trabajos

$ /usr/share/skype-wrapper/indicator-applet-skype.py
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
INFO: Attached complete
Traceback (most recent call last):
  File "/usr/share/skype-wrapper/indicator-applet-skype.py", line 832, in <module>
    skype = SkypeBehaviour();
  File "/usr/share/skype-wrapper/indicator-applet-skype.py", line 481, in __init__
    self.telepathy_presence = self.getPresence()
  File "/usr/share/skype-wrapper/indicator-applet-skype.py", line 793, in getPresence
    '/org/freedesktop/Telepathy/AccountManager')
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 244, in get_object
    follow_name_owner_changes=follow_name_owner_changes)
  File "/usr/lib/python2.7/dist-packages/dbus/proxies.py", line 246, in **init**
    self._named_service = conn.activate_name_owner(bus_name)
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 183, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 281, in start_service_by_name
    'su', (bus_name, flags)))
  File "/usr/lib/python2.7/dist-packages/dbus/connection.py", line 630, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Telepathy.AccountManager was not provided by any .service files

**Comment by sagetarian** [comment]
thanks for this .. i'll have a fix ready for you :) just to make sure, have you got telepathy-mission-control-5 installed?

skype-wrapper seems to think you do? i'll figure out a fix if you dont

**Comment by pablorq** [comment]
Wow! You've already have a fix in only 2 minutes! You're very fast!  ;D

telepathy-mission-control-5 1:5.9.1-0ubuntu2 is uninstalled. Shall it be installed?

**Comment by sagetarian** [comment]
https://github.com/downloads/sagetarian/skype-wrapper/skype-wrapper_0ubuntu5-4.1_all.deb

there we go :) telepathy-mission-control-5 is needed if you want to use the "presence" menu .. but this just checks if you have telepathy installed and also double checks if you have mission-control service running .. just out of curiosity in your terminal if you run

<code>dpkg -s telepathy-mission-control-5</code>

what does it print out?

**Comment by pablorq** [comment]
$ dpkg -s telepathy-mission-control-5
El paquete `telepathy-mission-control-5' no está instalado y no hay ninguna información disponible.
Utilice dpkg --info (= dpkg-deb --info) para examinar archivos,
y dpkg --contents (= dpkg-deb --contents) para listar su contenido.

$ sudo dpkg -s telepathy-mission-control-5
[sudo] password for parq: 
El paquete `telepathy-mission-control-5' no está instalado y no hay ninguna información disponible.
Utilice dpkg --info (= dpkg-deb --info) para examinar archivos,
y dpkg --contents (= dpkg-deb --contents) para listar su contenido.

In other words, that telepathy-mission-control-5 is not installed.

I'll be waiting if you need to do another tests, before to install 5-4.1.

**Comment by sagetarian** [comment]
ah okay :) that explains it .. i totally forgot about any other language than english :P i search for the text "not installed" in that.. but here's one with potential internationalization.. try running

<code>dpkg-query -l telepathy-mission-control-5</code>

and tell me the output

**Comment by pablorq** [comment]
$ dpkg-query -l telepathy-mission-control-5
Deseado=Desconocido/Instalar/Eliminar/Purgar/Retener
| Estado=No/Instalado/Config-files/Desempaquetado/Medio-conf/Medio-inst/espera-disparo/pendiente-disparo
|/ Err?=(ninguno)/Requiere-reinst (Estado,Err: mayúsc.=malo)
||/ Nombre         Versión       Descripción
+++-==============-==============-============================================
un  telepathy-miss <ninguna>      (no hay ninguna descripción disponible)

**Comment by sagetarian** [comment]
perfect :) i can use that reliably

**Comment by sagetarian** [comment]
the latest version is on its way to your PPA .. if you're impatient then download here

https://github.com/sagetarian/skype-wrapper/blob/master/skype-wrapper_0ubuntu5-4.3_all.deb?raw=true

let me know if this fixes it

**Comment by cmavr8** [comment]
No it doesn't!
I'm running 0ubuntu5-4.3~oneiric2 from your repository, while I have telepathy-mission-control-5 1:5.9.1-0ubuntu2 running but  I still get warnings identical to the above... (mainly "WARNING: Failed to attach skype-wrapper to Skype process")

And the app doesn't work..
Ubuntu 11.10 64-bit.
Please let me know if you need more info.
Thanks

**Comment by pablorq** [comment]
Updated from PPA to 0ubuntu5-4.3~oneiric2

Now it works well!

Good Job!

**Comment by troex** [comment]
Still unable to run

Ubuntu 11.10
Skype 2.2.0.35
skype-wrapper 0ubuntu5-4.3~oneiric

```
$ skype-wrapper --help
Starting skype-wrapper
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
```

```

$ dpkg-query -l telepathy-mission-control-5
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                 Version              Description
+++-====================-====================-========================================================
ii  telepathy-mission-co 1:5.9.1-0ubuntu2     management daemon for Telepathy real-time communication 
```

**Comment by chikamichi** [comment]
Hi, I have the same issue, on 11.10, using 0ubuntu5-4.3~oneiric2 and telepathy-mission-control-5 1:5.9.1-0ubuntu2. Launched skype (fine) then skype-wrapper:

```
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process
/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
^CTraceback (most recent call last):
  File "/usr/share/skype-wrapper/skype-wrapper.py", line 62, in <module>
    start_skype()
  File "/usr/share/skype-wrapper/skype-wrapper.py", line 30, in start_skype
    start_skype()
  File "/usr/share/skype-wrapper/skype-wrapper.py", line 30, in start_skype
    start_skype()
  File "/usr/share/skype-wrapper/skype-wrapper.py", line 30, in start_skype
    start_skype()
  File "/usr/share/skype-wrapper/skype-wrapper.py", line 28, in start_skype
    ret = subprocess.call(['python','indicator-applet-skype.py'])
  File "/usr/lib/python2.7/subprocess.py", line 493, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 1281, in wait
    pid, sts = _eintr_retry_call(os.waitpid, self.pid, 0)
  File "/usr/lib/python2.7/subprocess.py", line 478, in _eintr_retry_call
    return func(*args)
KeyboardInterrupt
```

**Comment by marcin-koziol** [comment]
I got this output when i removed the try...catch from indicator-applet-skype.py:

<pre>
INFO: Initializing Skype API
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
Traceback (most recent call last):
  File "indicator-applet-skype.py", line 857, in <module>
    skype = SkypeBehaviour();
  File "indicator-applet-skype.py", line 460, in __init__
    self.skype.Attach(Wait=True)
  File "/usr/lib/pymodules/python2.7/Skype4Py/skype.py", line 408, in Attach
    self._Api.attach(self.Timeout, Wait)
  File "/usr/lib/pymodules/python2.7/Skype4Py/api/posix_dbus.py", line 121, in attach
    self.start_watcher()
  File "/usr/lib/pymodules/python2.7/Skype4Py/api/posix_dbus.py", line 114, in start_watcher
    arg0='com.Skype.API')
  File "/usr/lib/python2.7/dist-packages/dbus/bus.py", line 151, in add_signal_receiver
    path, **keywords)
  File "/usr/lib/python2.7/dist-packages/dbus/connection.py", line 384, in add_signal_receiver
    self._require_main_loop()
RuntimeError: To make asynchronous calls, receive signals or export objects, D-Bus connections must be attached to a main loop by passing mainloop=... to the constructor or calling dbus.set_default_main_loop(...)

</pre>

**Comment by thomas-yi** [comment]
Hiya,

same problem here, I think:

`/usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion `g_type_from_name (name) == 0' failed
  import gobject._gobject
INFO: Initializing Skype API
INFO: Starting Skype
INFO: Waiting for Skype Process
INFO: Attaching skype-wrapper to Skype process
WARNING: Failed to attach skype-wrapper to Skype process`

Strange is, it was working 2 days ago! 

**edit:** actually it is working _sometimes_ ... like every 3rd time. At other times it's either not doing anything visible at all or filling up all the RAM. As this causes heavy swapping, it disables my system for 5+ minutes ...

T

**Comment by david-barbion** [comment]
Hi,

I was able to solve this by changing default transport to x11. Use the provided patch:

<pre>
--- /tmp/skype-wrapper/src/indicator-applet-skype.py    2012-04-07 03:10:26.000000000 +0200
+++ indicator-applet-skype.py   2012-04-26 23:59:35.771471298 +0200
@@ -518,7 +518,7 @@
   # initialize skype
   def __init__(self):
     log("Initializing Skype API", INFO)
-    self.skype = Skype4Py.Skype()
+    self.skype = Skype4Py.Skype(None, Transport='x11')
     self.call_ringing = 0
     self.calls_ringing_started = False
     
</pre>

**Comment by sagetarian** [comment]
i shall test that this works .. thanks for the patch .. if it works standard, i will include it in the next release

**Comment by sagetarian** [comment]
confirmed that it works on mine .. will create a dev .deb for these gents to try out .. thanks again for the patch

**Comment by sagetarian** [comment]
https://github.com/sagetarian/skype-wrapper/downloads

**Comment by sagetarian** [comment]
latest dev is 0.5.4.6

**Comment by sagetarian** [comment]
marking this one as fixed
